### PR TITLE
AEA-1238 filter config overrides

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

filter data for config manager overrides, to apply only updated values

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/configurations/manager.py [hunk 1]
```diff
@@ -498,6 +508,15 @@ def update_config(self, overrides: Dict) -> None:
 
         self.agent_config.update(overrides, env_vars_friendly=self.env_vars_friendly)
 
+    def _filter_overrides(self, overrides: Dict):
```
Comment (DavidMinarsch) [inline aea/configurations/manager.py:511]: please add typing of return type

File: aea/configurations/validation.py [hunk 2]
```diff
@@ -313,3 +313,31 @@ def check_excludes(path):
             )
 
     return errors
+
+
+SAME_MARK = object()
+
+
+def filter_data(base: Any, updates: Any) -> Any:
```
Comment (DavidMinarsch) [inline aea/configurations/validation.py:321]: Can we use type `JSONLike` instead (from `aea.common`)?

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Comment by DavidMinarsch** [comment]
Replaces #2126

File: aea/configurations/validation.py (hunk 2 above)
Comment (solarw) [reply aea/configurations/validation.py:321]: there is not only json friendly data, as i remember component_id also can be provided

File: aea/configurations/manager.py (hunk 1 above)
Comment (solarw) [reply aea/configurations/manager.py:511]: done

**Pull request merged**